src/scriptdev2/scripts/kalimdor/the_barrens.cpp

```diff
@@ -591,6 +592,123 @@ CreatureAI* GetAI_npc_wizzlecranks_shredder(Creature* pCreature)
     return new npc_wizzlecranks_shredderAI(pCreature);
 }
 
+/*#####
+## Quest Counterattack!
+QuestID 4,021
+#####*/
+enum
+{
+
+	NPC_KROMZAR = 9456,
+	NPC_KOLKAR_INVADER = 9524,
+	NPC_KOLKAR_STORMSEER = 9523,
+	NPC_HORDE_DEFENDER = 9457,
+	NPC_HORDE_AXE_THROWER = 9458,
+	NPC_REGTHAR_DEATHGATE = 3389,
+	QUEST_COUNTERATTACK = 4021,
+
+};
+
+struct npc_regthar_deathgateAI : public ScriptedAI {
+	npc_regthar_deathgateAI(Creature* pCreature) : ScriptedAI(pCreature) { Reset(); }
+
+
+
+
+	uint8 m_uiMobCount;
+
+	ObjectGuid m_playerGuid;
+
+	void Reset() override
+	{
```

Comment:
didnot notice this was a function define xD

doesn't make what i say invalid, just malplaced ^^